tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc

```diff
@@ -80,116 +85,126 @@ class MklConvBwdFilterPrimitive : public MklPrimitive {
  public:
   explicit MklConvBwdFilterPrimitive(
       const MklConvBwdFilterParams& convBwdFilterDims)
-      : cpu_engine_(engine::cpu, 0) {
-    context_.bwd_filter_stream.reset(new stream(stream::kind::eager));
-    // create conv primitive
+      : cpu_engine_(ENGINE_CPU, 0) {
+    context_.bwd_filter_stream.reset(new CPU_STREAM(cpu_engine_));
+
+    // Create convolution backward filter primitive.
     if (context_.conv_bwd_filter == nullptr) {
       Setup(convBwdFilterDims);
     }
   }
 
   ~MklConvBwdFilterPrimitive() {}
 
-  // Convolution backward weights with bias
-  //   src_data:         input data buffer of src
-  //   diff_filter_data: output data buffer of diff_filter
-  //   diff_bias_data:   output data buffer of diff_bias
-  //   diff_dst_data:    input data buffer of diff_dst
+  // Convolution backward weights execution with bias
+  //   src_data:         input data buffer for src
+  //   diff_filter_data: output data buffer for diff_filter
+  //   diff_bias_data:   output data buffer for diff_bias
+  //   diff_dst_data:    input data buffer for diff_dst
   void Execute(const T* src_data, const T* diff_filter_data,
                const T* diff_bias_data, const T* diff_dst_data) {
     context_.src_mem->set_data_handle(
         static_cast<void*>(const_cast<T*>(src_data)));
     context_.diff_filter_mem->set_data_handle(
         static_cast<void*>(const_cast<T*>(diff_filter_data)));
-    context_.diff_bias_mem->set_data_handle(
-        static_cast<void*>(const_cast<T*>(diff_bias_data)));
+    if (diff_bias_data != nullptr) {
+      context_.diff_bias_mem->set_data_handle(
+          static_cast<void*>(const_cast<T*>(diff_bias_data)));
+    }
     context_.diff_dst_mem->set_data_handle(
         static_cast<void*>(const_cast<T*>(diff_dst_data)));
 
+#ifdef ENABLE_MKLDNN_V1
+    DCHECK_EQ(context_.bwd_filter_primitives.size(),
+              context_.bwd_filter_primitives_args.size());
+    for (size_t i = 0; i < context_.bwd_filter_primitives.size(); ++i) {
+      context_.bwd_filter_primitives.at(i).execute(
+          *context_.bwd_filter_stream,
+          context_.bwd_filter_primitives_args.at(i));
+    }
```

Comment:
Please call `execute_primitives` instead.

Reply:
Sorry, somehow missed it. Fixed it.